packages/labs/task/src/test/task_test.ts

```diff
@@ -137,27 +103,129 @@ suite('Task', () => {
     }
   });
 
-  test('initial status', async () => {
-    el = new A();
-    assert.equal(el.task.value, undefined);
+  test('task without args do not run', async () => {
+    const el = await renderElement(getTestElement());
+    assert.equal(el.task.status, TaskStatus.INITIAL);
+    assert.equal(el.taskValue, undefined);
+    el.requestUpdate();
+    await tasksUpdateComplete();
     assert.equal(el.task.status, TaskStatus.INITIAL);
-    assert.equal(el.taskController.value, undefined);
-    assert.equal(el.taskController.status, TaskStatus.INITIAL);
+    assert.equal(el.taskValue, undefined);
   });
 
-  test('pending and complete status', async () => {
+  test('tasks with args run initially', async () => {
+    const el = getTestElement({args: () => [el.a, el.b]});
+    await renderElement(el);
     assert.equal(el.task.status, TaskStatus.PENDING);
-    assert.equal(el.taskController.status, TaskStatus.PENDING);
+    assert.equal(el.taskValue, undefined);
     el.resolveTask();
-    el.taskController.resolveTask();
     await tasksUpdateComplete();
     assert.equal(el.task.status, TaskStatus.COMPLETE);
-    assert.equal(el.taskController.status, TaskStatus.COMPLETE);
-    assert.equal(el.taskValue, 'result: foo, bar');
-    assert.equal(el.taskControllerValue, 'result: 0');
+    assert.equal(el.taskValue, `a,b`);
   });
 
-  test('error status', async () => {
+  test('tasks with args run when args change', async () => {
```

Comment:
It'd be good to also have a basic test that tasks don't run when args haven't changed.

Reply:
Added.